python/lsst/ctrl/execute/libexec/dagIdInfo.py

```diff
@@ -26,33 +26,35 @@
 import re
 import sys
 
+
 # extracts a line from a DAG file to show which ids were processed for a
 # particular dag node
-if __name__ == "__main__":
+def main():
     if len(sys.argv) != 3:
         print("usage:  %s dagNodeName filename" % os.path.basename(sys.argv[0]))
-        sys.exit(errno.EINVAL)
+        return errno.EINVAL
 
     dagNode = sys.argv[1]
     filename = sys.argv[2]
 
     if not os.path.exists(filename):
         print("file %s not found" % filename)
-        sys.exit(errno.ENOENT)
+        return errno.ENOENT
 
     ex = r"VARS %s var1=\"(?P<idlist>.+?)\"" % dagNode
-    file = open(filename)
-    for line in file:
-        line = line.rstrip(" \n")
-
-        # look for the line with the dagnode name in it
-        # and extract everything after "var1", but not the quotes
-        values = re.search(ex, line)
-        if values is None:
-            continue
-        ids = values.groupdict()["idlist"]
-        file.close()
-        print(ids)
-        sys.exit(0)
-    file.close()
-    sys.exit(0)
+    with open(filename) as file:
+        for line in file:
+            line = line.rstrip(" \n")
+
+            # look for the line with the dagnode name in it
+            # and extract everything after "var1", but not the quotes
+            values = re.search(ex, line)
+            if values is None:
+                continue
+            ids = values.groupdict()["idlist"]
+            print(ids)
+            break
+
```

Comment:
Do you want to be explicit and `return 0` here? Rather than ending up with implicit `sys.exit(None)`?

Reply:
No; `sys.exit(None)` is equivalent to `sys.exit(0)` so the implicit return is fine.